geopandas/tools/tests/test_sjoin.py

```diff
@@ -384,6 +391,22 @@ def test_sjoin_dwithin(
         joined = sjoin(left, right, how=how, predicate="dwithin", distance=distance)
         assert_frame_equal(expected_gdf.sort_index(), joined.sort_index())
 
+    # GH3239
+    def test_sjoin_left_within_order(self):
+        pts = GeoDataFrame(geometry=points_from_xy(*np.random.rand(2, 10)))
```

Comment:
Should fix this seed / replace with the output of running this. Unlikely to matter but probably best to avoid rng in the tests.

Reply:
With the non-random data, have you verified this test actually failed before? (that it doesn't have by accident the same order as it would be in the correct output)

Reply:
Good point. I did not. Will check as soon as I'm back home.

Reply:
Good call. Linspace wasn't testing the issue.